### PR TITLE
feat: Add artisan about command

### PR DESCRIPTION
**Pull request by almas1992**
## 📑 Description

This PR introduces an `artisan about` command to display system environment and application configuration details.

image

A new `About` function allows adding custom sections and entries to the about command output and can be easily called in the Boot method of a ServiceProvider for seamless integration.

```go
func (s *ServiceProvider) Boot() {
    facades.App().About("Custom", []foundation.AboutItem{
        {Key: "First Info", Value: "foo"},
    })
}

```

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `About` command to display application information.
	- Added structured handling for application-related details with the `AboutInfo` struct.

- **Bug Fixes**
	- Enhanced command registration process to include the new `About` command.

- **Tests**
	- Added unit tests for the `AboutCommand` functionality and related methods to ensure correct behavior.

- **Chores**
	- Updated mock capabilities for the `Application` type to support the new `About` method.

## ✅ Checks

- [x] Added test cases for my code

### CODE REVIEW

**Comment by almas1992** [comment]
Review Ready

File: console/console/about_command.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+		})
+	}
+	for i := range sections {
+		ranger(sections[i], info.details[info.section[sections[i]]])
+	}
+
+}
+
+// AddAboutInformation Add custom information to the application information.
+func AddAboutInformation(section, key, value string, more ...string) {
```
Comment (hwbrzzl) [inline console/console/about_command.go:152]: We can add a new method here: https://github.com/goravel/framework/blob/master/contracts/foundation/application.go#L31, and the variables can be `About(section, map[string]string)`

File: console/console/about_command.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+var customInformationResolvers []func()
+
+func NewAboutCommand(app foundation.Application) *AboutCommand {
+	return &AboutCommand{
+		app: app,
+	}
+}
+
+// Signature The name and signature of the console command.
+func (receiver *AboutCommand) Signature() string {
```
Comment (hwbrzzl) [inline console/console/about_command.go:38]: We use `r` uniformly.

File: console/console/about_command.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+		"Goravel Version", str.Of(receiver.app.Version()).LTrim("v").String(),
+		"Go Version", str.Of(runtime.Version()).LTrim("go").String(),
+		"Environment", configFacade.GetString("app.env"),
+		"Debug Mode", func() string {
+			if configFacade.GetBool("app.debug") {
+				return "<fg=yellow;op=bold>ENABLED</>"
+			}
+			return "OFF"
+		}(),
+		"URL", str.Of(configFacade.GetString("http.url")).Replace("http://", "").Replace("https://", "").String(),
```
Comment (hwbrzzl) [inline console/console/about_command.go:87]: Is the `Replace` required? If not, I think it's better to print `http.url` directly.

File: console/console/about_command.go [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+		"Queue", configFacade.GetString("queue.default"),
+		"Session", configFacade.GetString("session.driver"),
+	)
+	for i := range customInformationResolvers {
+		customInformationResolvers[i]()
+	}
+}
+
+// addToSection Add a new section to the application information.
+func (info *information) addToSection(section, key, vale string, more ...string) {
```
Comment (hwbrzzl) [inline console/console/about_command.go:115]: If the command is run multiple times in the compiled app, the same section will be duplicated.

File: console/console/about_command.go [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+		detail := kv{key: more[i]}
+		if i+1 < len(more) {
+			detail.value = more[i+1]
+		}
+		info.details[index] = append(info.details[index], detail)
+	}
+}
+
+// Range Iterate over the application information sections.
+func (info *information) Range(section string, ranger func(s string, details []kv)) {
```
Comment (hwbrzzl) [inline console/console/about_command.go:133]: Please add test cases for `addToSection` and `Range`.

**Review by hwbrzzl — COMMENTED** [review]
Great new command 👍

File: contracts/foundation/application.go [hunk 6]
```diff
@@ -28,6 +28,11 @@ import (
 	"github.com/goravel/framework/contracts/validation"
 )
 
+type AboutInfo struct {
```
Comment (hwbrzzl) [inline contracts/foundation/application.go:31]:
```suggestion
type AboutItem struct {
```

File: contracts/foundation/application.go [hunk 7]
```diff
@@ -63,6 +68,8 @@ type Application interface {
 	SetJson(json Json)
 	// GetJson get the JSON implementation.
 	GetJson() Json
+	// About add information to the application's about command.
+	About(section string, details []AboutInfo)
```
Comment (hwbrzzl) [inline contracts/foundation/application.go:72]:
```suggestion
About(section string, items []AboutItem)
```

**Review by hwbrzzl — COMMENTED** [review]
Great, only suggest modifying the variable names.

**Comment by almas1992** [comment]
Review Ready

**Review by hwbrzzl — APPROVED** [review]
Amazing

image

**Pull request merged**